src/Language/Haskell/HGrep/Print.hs

```diff
@@ -74,14 +77,14 @@ printSearchResult (PrintOpts co lno) (SearchResult anns ast) =
     wholeSrc :: [Char]
     wholeSrc = EP.exactPrint ast anns
 
-    nill, src :: [[Char]]
-    (nill, src) = L.span null $ L.lines wholeSrc
+    src :: [[Char]]
+    (_, src) = L.span (\s -> null s || all isSpace s) $ L.lines wholeSrc
 
     -- Doesn't prepent locations when there is no start line number
     printWithLineNums :: Maybe Int -> [Char]
-    printWithLineNums Nothing      = wholeSrc
+    printWithLineNums Nothing      = L.unlines src
```

Comment:
As `L.unlines src` is used in `where` block, probably it makes sense to move `nonNumberedSrc` also here? So can get rid off some duplicate code..